### ISSUE TITLE
[nodejs] Using unix timestamp for since does not work

### ISSUE THREAD

**Issue by danez** [facebook/watchman#392]
When I try to issue a subscribe query to watchman from nodejs with the since value being a number watchman throws an error.

```
Error: failed to parse query: invalid value for 'since'
    at BunserBuf.<anonymous> (node_modules/fb-watchman/index.js:95:23)
    at emitOne (events.js:96:13)
    at BunserBuf.emit (events.js:188:7)
    at BunserBuf.process (node_modules/bser/index.js:289:10)
    at node_modules/bser/index.js:244:12
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

The arguments for the subscribe query is:
```
{ 
  expression: [ 'allof', [ 'name', ["...", "long", "array", "of", "files", "..."], 'wholename' ] ],
  fields: [ 'name', 'mtime_ms', 'exists' ],
  since: 1480513073
 }
```

According to the doc and as far as I can see in [clockspec.cpp](https://github.com/facebook/watchman/blob/9619ff88d3504938e079a38c7778504cf6b66d06/clockspec.cpp#L29..L33) it should be allowed to send an integer.

Am I doing something wrong?

**Comment by danez** [comment]
Seems if I use `new Int64(startTime)` it works. Is that expected? Would be nice if that could be handled on the fly for numbers in nodejs.

**Comment by wez** [comment]
I don't see how that specific value would get encoded incorrectly.  The code tries to detect and encode based on what you pass in.  That specific value is < MAX_INT32 so it shouldn't need to be wrapped up in an Int64 wrapper.

Do your watchman logs give more context?  (I'm doubtful, but worth asking).

Are you definitely passing it as a number in the failing scenario?

Would you mind adding that number into the list here: https://github.com/facebook/watchman/blob/master/node/bser/test/bser.js#L41 and running the tests and see what happens?

**Comment by danez** [comment]
I'm definitely setting a number, because what I do is this: `since: Math.floor(startTime / 1000),` and this should always return an integer. 
Btw. watchman does not accept millisecond timestamps as a valid value for since by any way? Seconds seem not really accurate and a lot of things can happen in half a second.

But back to topic, I looked at the logs an found this message:

```
"log": "2016-11-30T17:16:45,239: [client=0x1ced200:stm=0x1aecee0] send_error_response: [\"subscribe\", \"...other stuff ... \", \"since\": 1480526199.0}] failed: failed to parse query: invalid value for 'since'\n",
```

Seems for some reason it is converted to a float. Going to debug why this happens.

**Comment by danez** [comment]
Ok found it. In bser.js [here](https://github.com/facebook/watchman/blob/master/node/bser/index.js#L492) every number is translated to double.

**Comment by wez** [comment]
ooof.

Would you mind submitting a pull request that uses `Number.isInteger` to gate calling into `dump_int` in the right case?

**Comment by danez** [comment]
Yes already working on it. It's just tricky to get the testcase right, as in js everything is a float, so `1.0 === 1` is `true`.

Also I wouldn't use `Number.isInteger` but rather the suggested check in the polyfill (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger) as `Number.isInteger()` is only available as of 0.12 and not sure what the support levels for the pser package are.

**Comment by wez** [comment]
I pushed out version 2.0.0 of fb-watchman and bser last week, sorry this took so long.
